mcsema/BC/Function.cpp

```diff
@@ -260,6 +303,123 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
   return block;
 }
 
+static void LiftExceptionFrameLP(TranslationContext &ctx,
+                                 const NativeFunction *cfg_func) {
+
+  if (cfg_func->eh_frame.size() > 0) {
+
+    // The personality function `__gxx_personality_v0` is lifted as global variable. Erase the
+    // variable before declaring it as the function.
+    if (auto personality_func_var = gModule->getGlobalVariable(PERSONALITY_FUNCTION)) {
+      personality_func_var->eraseFromParent();
+    }
+
+    auto personality = gModule->getFunction(PERSONALITY_FUNCTION);
+
+    if (personality == nullptr) {
+      personality = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt32Ty(*gContext), true),
+          llvm::Function::ExternalLinkage, PERSONALITY_FUNCTION, gModule);
+    }
+
+    auto lifted_func = gModule->getFunction(cfg_func->lifted_name);
+    lifted_func->addFnAttr(llvm::Attribute::UWTable);
+    lifted_func->addFnAttr(llvm::Attribute::OptimizeNone);
+    lifted_func->removeFnAttr(llvm::Attribute::NoUnwind);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+    lifted_func->setPersonalityFn(personality);
+#endif
+  }
+
+  for (auto &entry : cfg_func->eh_frame) {
+    if (!entry->lp_ea) {
+      continue;
+    }
+
+    if (!entry->action_index) {
+      std::stringstream ss;
+      ss << "cleanup_landingpad_" << std::hex << entry->lp_ea;
+      auto landing_bb = llvm::BasicBlock::Create(*gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
+                                            llvm::Type::getInt32Ty(*gContext), nullptr);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, 1, "cleanup.lpad");
+#else
+      auto personality = gModule->getFunction(PERSONALITY_FUNCTION);
+      auto lpad = ir.CreateLandingPad(exn_type, personality, 1, "cleanup.lpad");
+#endif
+      lpad->setCleanup(true);
+
+      auto exct_value_0  = ir.CreateExtractValue(lpad, 0);
+      auto store_loc_0 = ir.CreateAlloca(exct_value_0->getType());
+      ir.CreateStore(exct_value_0, store_loc_0);
+
+      auto exct_value_1 = ir.CreateExtractValue(lpad, 1);
+      auto store_loc_1 = ir.CreateAlloca(exct_value_1->getType());
+      ir.CreateStore(exct_value_1, store_loc_1);
+
+      // if `ctx.ea_to_block.count(entry->lp_ea) == 0`, the landing pad basic block
```

Comment:
Add an issue to track this and update the comment to reference the issue number.